source/geod24/concurrency.d

```diff
@@ -897,6 +898,78 @@ class FiberScheduler
         return ThreadInfo.thisInfo;
     }
 
+    final static class FiberBinarySemaphore
+    {
+
+        /***********************************************************************
+
+            Associate `FiberBinarySemaphore` with the running `Fiber`
+
+            `FiberScheduler` will check to see if the `Fiber` is blocking on a
+            `FiberBinarySemaphore` to avoid rescheduling it unnecessarily
+
+        ***********************************************************************/
+
+        private void registerToInfoFiber(InfoFiber info_fiber = cast(InfoFiber) Fiber.getThis()) nothrow
+        {
+            assert(info_fiber !is null, "This Fiber does not belong to FiberScheduler");
+            assert(info_fiber.sem is null, "This Fiber already has a registered FiberBinarySemaphore");
+            info_fiber.sem = this;
+
+        }
+
+        void wait() nothrow
+        {
+            this.registerToInfoFiber();
+            FiberScheduler.yield();
+            scope (exit) this.notified = false;
```

Comment:
I think this is in the wrong position. It should be placed at the start of the wait function.

Reply:
Yes, I will remove it. I know the order of scope statements matter when there is multiple. They get executed in reverse. But t